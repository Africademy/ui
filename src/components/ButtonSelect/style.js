import style from 'styled-components';

export const ButtonWrapper = style.div`
  display: flex;
`;
