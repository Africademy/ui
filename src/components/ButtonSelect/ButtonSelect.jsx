import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';
import Select from '../Select/Select';
import { ButtonWrapper } from './style';

/** Split Button component composed out of the Button and Select components.
 * The left side is the normal Button and the right side is the Select component that opens the menu popup on click. */
const ButtonSelect = ({
  onButtonClick, label, type, onSelectClick, items,
}) => (
  <ButtonWrapper>
    <Button isSplit onClick={onButtonClick} label={label} type={type}>{label}</Button>
    <Select isSplit type={type} onSelectClick={onSelectClick} items={items} />
  </ButtonWrapper>
);

ButtonSelect.propTypes = {
  /** Function to call when the left side, the Button is clicked */
  onButtonClick: PropTypes.func.isRequired,

  /** Label to show on the left Button side */
  label: PropTypes.string.isRequired,

  /** Type of the Button, Primary of Secondary */
  type: PropTypes.oneOf(['primary', 'secondary']),

  /** Function to call when the Select menu item is clicked */
  onSelectClick: PropTypes.func.isRequired,

  /** Items to show in the Select menu popup */
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  })).isRequired,
};

ButtonSelect.defaultProps = {
  type: 'secondary',
};

export default ButtonSelect;
