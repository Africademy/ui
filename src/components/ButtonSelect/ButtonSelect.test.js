import snap from 'jest-auto-snapshots';
import ButtonSelect from './ButtonSelect';

snap(ButtonSelect, './ButtonSelect.jsx');
