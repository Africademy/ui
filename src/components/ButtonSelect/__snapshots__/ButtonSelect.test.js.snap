// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > ButtonSelect Matches snapshot when array prop "items" has one item: "[object Object]" 1`] = `
<ForwardRef>
  <Button
    disabled={false}
    isSplit={true}
    label="jest-auto-snapshots String Fixture"
    loading={false}
    onClick={[MockFunction]}
    size="medium"
    type="primary"
  >
    jest-auto-snapshots String Fixture
  </Button>
  <Select
    isSplit={true}
    items={
      Array [
        Object {
          "id": "jest-auto-snapshots String Fixture",
          "title": "jest-auto-snapshots String Fixture",
        },
      ]
    }
    label=""
    onSelectClick={[MockFunction]}
    type="primary"
  />
</ForwardRef>
`;

exports[`jest-auto-snapshots > ButtonSelect Matches snapshot when passed all props 1`] = `
<ForwardRef>
  <Button
    disabled={false}
    isSplit={true}
    label="jest-auto-snapshots String Fixture"
    loading={false}
    onClick={[MockFunction]}
    size="medium"
    type="primary"
  >
    jest-auto-snapshots String Fixture
  </Button>
  <Select
    isSplit={true}
    items={
      Array [
        Object {
          "id": "jest-auto-snapshots String Fixture",
          "title": "jest-auto-snapshots String Fixture",
        },
      ]
    }
    label=""
    onSelectClick={[MockFunction]}
    type="primary"
  />
</ForwardRef>
`;

exports[`jest-auto-snapshots > ButtonSelect Matches snapshot when passed only required props 1`] = `
<ForwardRef>
  <Button
    disabled={false}
    isSplit={true}
    label="jest-auto-snapshots String Fixture"
    loading={false}
    onClick={[MockFunction]}
    size="medium"
    type="secondary"
  >
    jest-auto-snapshots String Fixture
  </Button>
  <Select
    isSplit={true}
    items={
      Array [
        Object {
          "id": "jest-auto-snapshots String Fixture",
          "title": "jest-auto-snapshots String Fixture",
        },
      ]
    }
    label=""
    onSelectClick={[MockFunction]}
    type="secondary"
  />
</ForwardRef>
`;
